### PR TITLE
kvserver/rangefeed: metamorphically enable RangefeedUseBufferedSender

### PR DESCRIPTION
**Pull request by wenyihu6**
**kvserver/rangefeed: rename .buffered_stream_sender. to .buffered_sender.**

This patch renames kv.rangefeed.buffered_stream_sender.enabled to
kv.rangefeed.buffered_sender.enabled to align with variable and
struct names better.

Epic: none
Release: none

---

**kvserver/rangefeed: metamorphically enable RangefeedUseBufferedSender**

This patch metamorphically enables the cluster setting
RangefeedUseBufferedSender.

Release note: none
Part of: https://github.com/cockroachdb/cockroach/issues/135332

### CODE REVIEW

**Comment by wenyihu6** [comment]
How do we feel about lifting the restriction here https://github.com/cockroachdb/cockroach/blob/a9076591f5bd9841040fea644bca63465103fa3a/pkg/kv/kvserver/replica_rangefeed.go#L102-L107?

File: pkg/kv/kvserver/replica_rangefeed.go [hunk 1]
```diff
@@ -95,10 +96,10 @@ var ErrBufferedSenderNotSupported = unimplemented.NewWithIssue(
 // unimplemented and disabled everywhere (#126560).
```
Comment (stevendanna) [inline pkg/kv/kvserver/replica_rangefeed.go:96]: Let's update the comment while we are here. This is no longer unimplemented :D.

**Review by stevendanna — APPROVED** [review]
Thanks!

File: pkg/kv/kvserver/replica_rangefeed.go (hunk 1 above)
Comment (wenyihu6) [reply pkg/kv/kvserver/replica_rangefeed.go]: 🥳 Good catch, done.

**Comment by wenyihu6** [comment]
TFTR!

bors r=stevendanna

**Comment by wenyihu6** [comment]
Extended CI failure TestLogGrowthWhenRefreshingPendingCommands looks unrelated.

**Pull request merged**